src/mesh/generic/Boundary.cpp

```diff
@@ -109,19 +114,17 @@ void Boundary::generate_sinusoidal(json const& boundary,
 
         auto const scale_factor = number_of_cycles[i] * period[i] / time_block.back();
 
-        time_block = time_block | view::transform([scale_factor, new_times](double x) {
-                         return x * scale_factor + new_times.back();
+        time_block = time_block | view::transform([scale_factor, times](double x) {
+                         return x * scale_factor + times.back();
                      });
 
-        new_times.insert(new_times.end(), time_block.begin(), time_block.end());
+        times.insert(times.end(), time_block.begin(), time_block.end());
 
-        new_loads = new_times | view::transform([i, amplitude, omega, phase](double x) {
-                        return amplitude[i] * std::sin(omega * x + phase[i]);
-                    });
+        loads = times | view::transform([i, amplitude, omega, phase](double x) {
```

Comment:
Perhaps capture by reference for amplitude and phase:
```cpp
&amplitude, ..., &phase](
```
to avoid a copy.

Reply:
One day you won't need to tell me about references and const :)